extensions/oidc-client/runtime/src/main/java/io/quarkus/oidc/client/OidcClient.java

```diff
@@ -33,14 +33,38 @@ default Uni<Tokens> getTokens() {
      * @param refreshToken refresh token
      * @return Uni<Tokens>
      */
-    Uni<Tokens> refreshTokens(String refreshToken);
+    default Uni<Tokens> refreshTokens(String refreshToken) {
+        return refreshTokens(refreshToken, Collections.emptyMap());
+    }
+
+    /**
+     * Refresh and return a new pair of access and refresh tokens.
+     * Note a refresh token grant will typically return not only a new access token but also a new refresh token.
+     *
+     * @param refreshToken refresh token
+     * @param additionalGrantParameters additional grant parameters
+     * @return Uni<Tokens>
+     */
+    Uni<Tokens> refreshTokens(String refreshToken, Map<String, String> additionalGrantParameters);
 
     /**
      * Revoke the access token.
      *
-     * @param refreshToken access token which needs to be revoked
+     * @param accessToken access token which needs to be revoked
+     * @return Uni<Boolean> true if the token has been revoked or found already being invalidated,
+     *         false if the token can not be currently revoked in which case a revocation request might be retried.
+     */
+    default Uni<Boolean> revokeAccessToken(String accessToken) {
+        return revokeAccessToken(accessToken, Collections.emptyMap());
+    }
+
+    /**
+     * Revoke the access token.
+     *
+     * @param accessToken access token which needs to be revoked
+     * @param additionalGrantParameters additional grant parameters
```

Comment:
@Eng-Fouad Can you please change the name of the parameter to `additionalParameters` as revoking  a token is not a grant request.